.github/workflows/ci.yml

```diff
@@ -36,3 +36,34 @@ jobs:
         cache: true
     - name: Run tests
       run: make test
+
+  run-benchdiff:
+    runs-on: ubuntu-latest
+    permissions:
+      checks: write
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-go@v4
+      with:
+        go-version-file: go.mod
+    # Version: https://github.com/WillAbides/benchdiff-action/releases/tag/v0.3.3
+    - uses: WillAbides/benchdiff-action@4d1d267fa96763646dd7c0d58e242817ce392c61
+      id: benchdiff
+      with:
+        benchdiff_version: 0.9.1
+        status_sha: ${{ github.sha }}
+        status_name: benchdiff-result
+        status_on_degraded: neutral
+        # See https://github.com/WillAbides/benchdiff
+        benchdiff_args: |
+          --base-ref=origin/main
+          --cpu=1,2
+          --count=5
+          --warmup-count=1
+          --warmup-time=10ms
+          --benchtime=100ms
+          --tolerance=20
```

Comment:
>   --tolerance=20     The minimum percent change before a result is considered degraded.

In this particular case if 20% variation then it will report a degradation.

For further details please refers to https://github.com/WillAbides/benchdiff#benchdiff---help-output